TFT/src/User/Menu/Heat.c

```diff
@@ -266,20 +266,24 @@ void menuHeat(void)
     update_time=300;
 }
 
+u32 lastHeatCheckTime = 0;
+void updateLastHeatCheckTime(void)
+{
+  lastHeatCheckTime = OS_GetTime();
+}
+
 
 void loopCheckHeater(void)
 {
   u8 i;
-  static u32  nowTime=0;
 
   do
   {  /* Send M105 query temperature continuously	*/
-    if(update_waiting == true)                {nowTime=OS_GetTime();break;}
-    if(OS_GetTime()<nowTime+update_time)       break;
+    if(update_waiting == true)                {updateLastHeatCheckTime();break;}
+    if(OS_GetTime() - lastHeatCheckTime < update_time)       break;

```

Comment:
FYI, when comparing two unsigned values like this, it's much safer to subtract the larger from the smaller and then compare to the delta, as I've done here. That will avoid any potential problems due to accidental comparison with signed values or due to integer overflow. This way is ALWAYS guaranteed to work the way you think it will.